lib/nostrum/api.ex

```diff
@@ -725,6 +725,9 @@ defmodule Nostrum.Api do
   Deletes multiple messages from a channel.
 
   `messages` is a list of `Nostrum.Struct.Message.id` that you wish to delete.
+  When given more than 100 messages, this function will chunk the given message
+  list into blocks of 100 and send them off to the API. It will stop deleting
+  on the first error that occurs.
```

Comment:
We should have a note here that deleting thousands of messages will take a pretty long time, and it some cases it might be more proper to just delete the channel and recreate it.

Reply:
I'm not sure if this has also been considered:

Only messages within the past two weeks can be deleted. Using this method (with large `messages`) may incur batches of history that mix snowflakes that are both within and outside of this window, so it may fail to bulk delete messages that can, in fact, be deleted.

In `discordrb` we synthesize a reference snowflake value to filter the users input. ([implementation](https://github.com/meew0/discordrb/blob/master/lib/discordrb/data.rb#L1997-L2015))

So:
- Do you expect the user to always pass suitable snowflakes?
- Do you want to ignore them internally (or make it optionable)

Not only for long-reaching bulk-deletes, but users attempting to prune "stale" channels (the latest message is already >2 weeks old) may get bitten by this.

---

[Relevant docs](https://discordapp.com/developers/docs/resources/channel#bulk-delete-messages)

Example API error:
```
$ curl \
  -H 'Content-Type: application/json' \
  -H 'Authorization: Bot TOKEN' \
  --data '{"messages":["461221076626833430", "427681027134324736", "427681350116704268"]}' \
  -X POST https://discordapp.com/api/v6/channels/246283902652645376/messages/bulk-delete
```
```
{"code": 50034, "message": "You can only bulk delete messages that are under 14 days old."}
```
`46*` is less than a day old, and `42*` are about 3 months old.

Reply:
Added in b0cb940401061ec4114a2497f4108caca5c1579e